polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -1710,211 +1710,6 @@ ValueCategory MLIRScanner::VisitBinaryOperator(clang::BinaryOperator *BO) {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result;
-    if (prev.getType().isa<mlir::FloatType>()) {
-      mlir::Value val = rhs.getValue(builder);
-      auto prevTy = val.getType().cast<mlir::FloatType>();
-      auto postTy = prev.getType().cast<mlir::FloatType>();
-
-      if (prevTy.getWidth() < postTy.getWidth()) {
-        val = builder.create<arith::ExtFOp>(loc, postTy, val);
-      } else if (prevTy.getWidth() > postTy.getWidth()) {
-        val = builder.create<arith::TruncFOp>(loc, postTy, val);
-      }
-      result = builder.create<arith::DivFOp>(loc, prev, val);
-    } else {
-      if (signedType)
-        result =
-            builder.create<arith::DivSIOp>(loc, prev, rhs.getValue(builder));
-      else
-        result =
-            builder.create<arith::DivUIOp>(loc, prev, rhs.getValue(builder));
-    }
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_ShrAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result;
-
-    if (signedType)
-      result = builder.create<ShRSIOp>(loc, prev, rhs.getValue(builder));
-    else
-      result = builder.create<ShRUIOp>(loc, prev, rhs.getValue(builder));
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_ShlAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result =
-        builder.create<ShLIOp>(loc, prev, rhs.getValue(builder));
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_RemAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result;
-
-    if (prev.getType().isa<mlir::FloatType>()) {
-      result = builder.create<RemFOp>(loc, prev, rhs.getValue(builder));
-    } else {
-      if (signedType)
-        result = builder.create<RemSIOp>(loc, prev, rhs.getValue(builder));
-      else
-        result = builder.create<RemUIOp>(loc, prev, rhs.getValue(builder));
-    }
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_AndAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result =
-        builder.create<AndIOp>(loc, prev, rhs.getValue(builder));
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_OrAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result =
-        builder.create<OrIOp>(loc, prev, rhs.getValue(builder));
-    lhs.store(builder, result);
-    return lhs;
-  }
-  case clang::BinaryOperator::Opcode::BO_XorAssign: {
-    assert(lhs.isReference);
-    auto prev = lhs.getValue(builder);
-
-    mlir::Value result =
-        builder.create<XOrIOp>(loc, prev, rhs.getValue(builder));
-    lhs.store(builder, result);
-    return lhs;
-  }
-
```

Comment:
As it can be seen, this code was:
1. Not reusing the code for the underlying operators
2. Returning the LHS instead of the computed value